Tasks/Common/sonarqube-common/sonarqube-common.ts

```diff
@@ -39,26 +46,41 @@ export function applySonarQubeAnalysisParams(toolRunner:ToolRunner, sqProjectNam
     return toolRunner;
 }
 
-export class SonarQubeEndpoint {
-    constructor(public Url, public Username, public Password) {
+// Run SQ analysis in issues mode, but only in PR builds
+export function applySonarQubeIssuesModeInPrBuild(toolrunner: ToolRunner) {
+
+    if (isPrBuild()) {
+        console.log(tl.loc('sqAnalysis_IncrementalMode'));
+
+        toolrunner.arg("-Dsonar.analysis.mode=issues");
+        toolrunner.arg("-Dsonar.report.export.path=sonar-report.json");
     }
+    else
+    {
+        tl.debug("Running a full SonarQube analysis");
+    }
+
+    return toolrunner;
 }
 
 // Gets SonarQube connection endpoint details.
-export function getSonarQubeEndpointFromInput(inputFieldName):SonarQubeEndpoint {
+export function getSonarQubeEndpointFromInput(inputFieldName): SonarQubeEndpoint {
     var errorMessage = "Could not decode the generic endpoint. Please ensure you are running the latest agent (min version 0.3.2)";
     if (!tl.getEndpointUrl) {
-        throw new Error(errorMessage);
+        tl.error(errorMessage);
+        tl.exit(1);
```

Comment:
I would be careful of this, I know that in at least one location the Maven task tries to hold off on the tl.exit(1) call until later so that code coverage results can be uploaded.